packages/SwingSet/src/kernel/state/vatKeeper.js

```diff
@@ -39,11 +39,30 @@ const FIRST_DEVICE_ID = 70n;
  * @param {*} transcriptStore  Accompanying transcript store
  * @param {string} vatID The vat ID string of the vat in question
  * TODO: consider making this part of makeVatKeeper
+ * @param {SourceOfBundle} source
+ * @param {RecordedVatOptions} options
  */
-export function initializeVatState(kvStore, transcriptStore, vatID) {
+export function initializeVatState(
+  kvStore,
+  transcriptStore,
+  vatID,
+  source,
+  options,
+) {
+  assert(options.workerOptions, `vat ${vatID} options missing workerOptions`);
+  assert(source);
+  assert('bundle' in source || 'bundleName' in source || 'bundleID' in source);
+  assert.typeof(options, 'object');
```

Comment:
aye, I'll change the order of the asserts

I guess I could factor out a two-line function (plus some asserts), and call it from both places, but it didn't seem worth it at the time, and I didn't think of a good name for it. `doSetSourceAndOptions(kvStore, source, options)` ?

It's important that `initializeVatState()` remain a standalone function, not a method on `vatKeeper`, because it must be called before creating a `vatKeeper()` (which asserts the presence of `.options`, to populate the `reapDirtThreshold` RAM cache).

I'll see what I can do.